app/Controller/Component/Auth/ApacheAuthenticate.php

```diff
@@ -73,6 +73,13 @@ public function authenticate(CakeRequest $request, CakeResponse $response)
         ldap_set_option($ldapconn, LDAP_OPT_PROTOCOL_VERSION, Configure::read('ApacheSecureAuth.ldapProtocol'));
         ldap_set_option($ldapconn, LDAP_OPT_REFERRALS, Configure::read('ApacheSecureAuth.ldapAllowReferrals', true));
 
+        if (Configure::read('ApacheSecureAuth.starttls', false) === true) {
```

Comment:
I'd use
```php
if (!empty(Configure::read('ApacheSecureAuth.starttls'))) {
```
If someone configure the entry like this
```php
'ApacheSecureAuth.starttls' => 1
```
It will be parsed as false

Reply:
But, then you would not be able to (technically) turn off starttls without removeing the setting?

Reply:
But i should change from === to ==, then both true, 1, "-1", etc would work. https://www.php.net/manual/en/types.comparisons.php

Reply:
Force pushed the above change.